### PR TITLE
Fix #15383 drop PJAX

### LINKED ISSUE #15383: Drop PJAX
- [ ] Remove PJAX itself.
- [ ] Remove PJAX-specific headers usage.

### PR DESCRIPTION
**Pull request by klimov-paul**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | yes
| Tests pass?   | yes
| Fixed issues  | #15383

### CODE REVIEW

File: framework/web/Response.php [hunk 1]
```diff
@@ -890,11 +890,7 @@ public function redirect($url, $statusCode = 302, $checkAjax = true)
                     // Ajax 302 redirect in IE does not work. Change status code to 200. See https://github.com/yiisoft/yii2/issues/9670
                     $statusCode = 200;
                 }
-                if (Yii::$app->getRequest()->getIsPjax()) {
-                    $this->setHeader('X-Pjax-Url', $url);
-                } else {
-                    $this->setHeader('X-Redirect', $url);
-                }
+                $this->setHeader('X-Redirect', $url);
```
Comment (samdark) [inline framework/web/Response.php]: I think this one was for PJAX as well.

Comment (klimov-paul) [reply framework/web/Response.php]: Why?
The original condition was:

```
if (Yii::$app->getRequest()->getIsPjax()) {
    $this->setHeader('X-Pjax-Url', $url);
} else {
    $this->setHeader('X-Redirect', $url);
}
```

check for AJAX is located above that.

Comment (samdark) [reply framework/web/Response.php]: Ah, OK.

Comment (githubjeka) [reply framework/web/Response.php]: http://www.yiiframework.com/doc-2.0/yii-web-response.html#redirect()-detail

**Review by samdark — APPROVED** [review]

File: framework/UPGRADE.md [hunk 2]
```diff
@@ -75,6 +75,8 @@ Upgrade from Yii 2.0.x
   Include it in your composer.json if you use it.
 * Masked input field widget was moved into separate extension https://github.com/yiisoft/yii2-maskedinput.
   Include it in your composer.json if you use it.
+* PJAX support has been removed: widget `yii\widget\Pjax`, method `yii\web\Request::getIsPjax()`, PJAX related checks and
+  headers are no longer available. If you wish to use PJAX further you should create your own integration for it.
```
Comment (githubjeka) [inline framework/UPGRADE.md]: > you should create your own integration for it. 

:( severely

Comment (githubjeka) [reply framework/UPGRADE.md]: https://github.com/yiisoft/yii2/issues/15383#issuecomment-352716024
Ah, OK

**Comment by jdde** [comment]
Hi guys, we're using pjax in our Project in yii2.
So when you drop pjax, what is the alternative to use? For example to refresh Gridviews inpage?

Greetz

**Comment by githubjeka** [comment]
@jdde see Fixed issues  #15383

**Pull request merged**